From b44461c099e47f46e3503999660be9300d297010 Mon Sep 17 00:00:00 2001
From: "hayim@marvell.com" <hayim@marvell.com>
Date: Thu, 25 Feb 2016 19:24:20 +0200
Subject: [PATCH 030/239] apn806: add rfu and ccu windows initialization

- Those configurations were added to 7040-rz and
  now also to apn806
- IOB configuration was not added since it's only
  for CP

Change-Id: Ibf4b12780d1c9ba87ee6b77202985b83790a0264
Signed-off-by: hayim@marvell.com <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27895
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 plat/marvell/a8k/apn806/plat_bl31_setup.c | 45 +++++++++++++++++++++++++++++++
 plat/marvell/a8k/apn806/platform.mk       | 35 +++++++++++++++---------
 2 files changed, 68 insertions(+), 12 deletions(-)
 create mode 100644 plat/marvell/a8k/apn806/plat_bl31_setup.c

diff --git a/plat/marvell/a8k/apn806/plat_bl31_setup.c b/plat/marvell/a8k/apn806/plat_bl31_setup.c
new file mode 100644
index 0000000..cae9a68
--- /dev/null
+++ b/plat/marvell/a8k/apn806/plat_bl31_setup.c
@@ -0,0 +1,45 @@
+/*
+* ***************************************************************************
+* Copyright (C) 2016 Marvell International Ltd.
+* ***************************************************************************
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* Neither the name of Marvell nor the names of its contributors may be used
+* to endorse or promote products derived from this software without specific
+* prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+***************************************************************************
+*/
+
+#include <plat_marvell.h>
+#include <apn806_setup.h>
+
+/* This function overruns the same function in marvell_bl31_setup.c */
+void bl31_plat_arch_setup(void)
+{
+	marvell_bl31_plat_arch_setup();
+
+	/* configure apn806 */
+	apn806_init();
+}
diff --git a/plat/marvell/a8k/apn806/platform.mk b/plat/marvell/a8k/apn806/platform.mk
index 065aba1..926aa28 100644
--- a/plat/marvell/a8k/apn806/platform.mk
+++ b/plat/marvell/a8k/apn806/platform.mk
@@ -35,11 +35,13 @@ PLAT_SRC			:= a7040_rz
 PLAT_FAMILY			:= a8k
 PLAT_FAMILY_BASE	:= plat/marvell/$(PLAT_FAMILY)
 PLAT_SRC_BASE		:= $(PLAT_FAMILY_BASE)/$(PLAT_SRC)
+PLAT_APN_SRC_BASE	:= $(PLAT_FAMILY_BASE)/$(PLAT)
 PLAT_INCLUDE_BASE	:= include/plat/marvell/$(PLAT_FAMILY)
+MARVELL_DRV_BASE	:= drivers/marvell
 
 CALL_DOIMAGE		:= y
 # This define is used in BL31 linker script to make sure
-# the PSCI code is aligned to 64KB
+# the PSCI code is alligned to 64KB
 $(eval $(call add_define,PLAT_MARVELL_7040_RZ1))
 
 MARVELL_GIC_SOURCES	:=	drivers/arm/gic/common/gic_common.c	\
@@ -49,12 +51,14 @@ MARVELL_GIC_SOURCES	:=	drivers/arm/gic/common/gic_common.c	\
 
 ATF_INCLUDES		:=	-Iinclude/common/tbbr
 
-PLAT_INCLUDES		:=	-I$(PLAT_FAMILY_BASE)/$(PLAT_SRC)			\
-				-I$(PLAT_FAMILY_BASE)/$(PLAT_SRC)/include			\
-				-I$(PLAT_INCLUDE_BASE)/$(PLAT_SRC)/board/common		\
-				-I$(PLAT_INCLUDE_BASE)/common					\
-				-I$(PLAT_INCLUDE_BASE)/common/aarch64			\
-				-I$(PLAT_SRC_BASE)								\
+PLAT_INCLUDES		:=	-I$(PLAT_FAMILY_BASE)/$(PLAT_SRC)		\
+				-I$(PLAT_FAMILY_BASE)/$(PLAT_SRC)/include	\
+				-I$(PLAT_INCLUDE_BASE)/$(PLAT_SRC)/board/common	\
+				-I$(PLAT_INCLUDE_BASE)/$(PLAT_SRC)/board	\
+				-I$(PLAT_INCLUDE_BASE)/common			\
+				-I$(PLAT_INCLUDE_BASE)/common/aarch64		\
+				-I$(PLAT_SRC_BASE)				\
+				-Iinclude/drivers/marvell			\
 				${ATF_INCLUDES}
 
 PLAT_BL_COMMON_SOURCES	:=	$(PLAT_SRC_BASE)/aarch64/a7040_rz_common.c	\
@@ -64,17 +68,24 @@ PLAT_BL_COMMON_SOURCES	:=	$(PLAT_SRC_BASE)/aarch64/a7040_rz_common.c	\
 BL1_SOURCES		+=	$(PLAT_SRC_BASE)/aarch64/plat_helpers.S	\
 				lib/cpus/aarch64/cortex_a72.S
 
-BL31_SOURCES		+=	lib/cpus/aarch64/cortex_a72.S		\
-				$(PLAT_SRC_BASE)/aarch64/plat_helpers.S		\
-				$(PLAT_SRC_BASE)/aarch64/plat_psci.S		\
-				$(PLAT_SRC_BASE)/plat_pm.c					\
+MARVELL_DRV		:= 	$(MARVELL_DRV_BASE)/rfu.c	\
+				$(MARVELL_DRV_BASE)/ccu.c
+
+BL31_SOURCES		+=	lib/cpus/aarch64/cortex_a72.S				\
+				$(PLAT_SRC_BASE)/aarch64/plat_helpers.S			\
+				$(PLAT_SRC_BASE)/aarch64/plat_psci.S			\
+				$(PLAT_SRC_BASE)/plat_pm.c				\
 				$(PLAT_SRC_BASE)/plat_topology.c			\
+				$(PLAT_APN_SRC_BASE)/plat_bl31_setup.c			\
+				$(PLAT_SRC_BASE)/apn806_setup.c				\
 				plat/marvell/common/marvell_gicv2.c			\
+				$(PLAT_SRC_BASE)/board/marvell_plat_config.c		\
+				${MARVELL_DRV}						\
 				${MARVELL_GIC_SOURCES}
 
 # Disable the PSCI platform compatibility layer (allows porting from Old Platform APIs
 # to the new APIs).
-# It is not needed since Marvell platform already uses the new platform APIs.
+# It is not needed since Marvell platform already used the new platform APIs.
 ENABLE_PLAT_COMPAT	:= 	0
 
 include plat/marvell/common/marvell_common.mk
-- 
1.9.1

